src/app/grid/grid-search-sample/grid-search-sample.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit, ViewChild, ViewEncapsulation } from "@angular/core";
+import { IgxColumnComponent } from "igniteui-angular/grid/column.component";
+import { IgxGridComponent } from "igniteui-angular/grid/grid.component";
+import { MARKET_DATA } from "./data";
+
+@Component({
+    selector: "grid-search-sample",
+    styleUrls: ["./grid-search-sample.component.scss"],
+    templateUrl: "./grid-search-sample.component.html"
+})
+export class GridSearchSampleComponent implements OnInit {
+
+    @ViewChild("grid1") public grid: IgxGridComponent;
+    public data: any[];
+    public searchText: string = "";
+    public caseSensitive: boolean = false;
+
+    public ngOnInit(): void {
+        this.data = MARKET_DATA;
+    }
+
+    public clearSearch() {
+        this.searchText = "";
+        this.grid.clearSearch();
+    }
+
+    public searchKeyDown(ev) {
+        if (ev.key === "Enter" || ev.key === "ArrowDown" || ev.key === "ArrowRight") {
+            this.grid.findNext(this.searchText, this.caseSensitive);
```

Comment:
Add ev.preventDefault(); calls before findNext and findPrev methods, this will prevent the caret from moving when you are navigating the search results.